mempool/mempool.go

```diff
@@ -191,3 +191,29 @@ func (e ErrPreCheck) Error() string {
 func IsPreCheckError(err error) bool {
 	return errors.As(err, &ErrPreCheck{})
 }
+
+var ErrTxNotFound = errors.New("transaction not found in mempool")
+
+type ErrCheckTxAsync struct {
+	Reason error
+}
+
+func (e ErrCheckTxAsync) Error() string {
+	return e.Reason.Error()
+}
+
+type ErrAppConnMempool struct {
+	Reason error
+}
+
+func (e ErrAppConnMempool) Error() string {
+	return e.Reason.Error()
+}
+
+type ErrFlushAppConn struct {
+	Reason error
+}
+
+func (e ErrFlushAppConn) Error() string {
+	return e.Reason.Error()
+}
```

Comment:
I think it is best to define these errors in `errors.go` file to follow the standard

Reply:
Agreed!